src/devices/SensorHub/SensorHub.cs

```diff
@@ -0,0 +1,183 @@
+            _i2cDevice = i2cDevice;
+        }
+
+        /// <summary>
+        /// Try to read the temperature from the off board thermometer in the range -30 to 127 Celsius
+        /// </summary>
+        /// <param name="temperature">The temperature in Celsius if successful</param>
+        /// <returns></returns>
+        public bool TryReadOffBoardTemperature(out Temperature temperature)
+        {
+            temperature = Temperature.MaxValue;
+            var status = ReadRegister(Register.STATUS_REG);
+            if (!((status & 0x1) == 0x1) && !((status & 0x2) == 0x2))
+            {
+                temperature = Temperature.FromDegreesCelsius(ReadRegister(Register.TEMP_REG));
+                return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Try to read the temperature from the barometer sensor in the range -40 to 80 Celsius
+        /// </summary>
+        /// <param name="temperature">Temperature in Celsius if successful</param>
+        /// <returns></returns>
+        public bool TryReadBarometerTemperature(out Temperature temperature)
+        {
+            temperature = Temperature.MaxValue;
+            var status = ReadRegister(Register.BMP280_STATUS);
+            if (status == 0)
+            {
+                temperature = Temperature.FromDegreesCelsius(ReadRegister(Register.BMP280_TEMP_REG));
+                return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Try to read pressure in Pascals from sensor in the range 300 Pa to 1100 hPa
+        /// </summary>
+        /// <param name="pressure">pressure in Pascals if successful</param>
+        /// <returns></returns>
+        public bool TryReadBarometerPressure(out Pressure pressure)
+        {
+            pressure = Pressure.MaxValue;
+            var status = ReadRegister(Register.BMP280_STATUS);
+            if (status == 0)
+            {
+                pressure = Pressure.FromPascals(
+                       ReadRegister(Register.BMP280_PRESSURE_REG_L)
+                    | (ReadRegister(Register.BMP280_PRESSURE_REG_M) << 8)
+                    | (ReadRegister(Register.BMP280_PRESSURE_REG_H) << 16));
+                return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Try to read illuminance in Lux from sensor in the range 0lx to 1800lx
+        /// </summary>
+        /// <param name="illuminance">Illuminance in Lux if successful</param>
+        /// <returns></returns>
+        public bool TryReadIlluminance(out Illuminance illuminance)
+        {
+            illuminance = Illuminance.MaxValue;
+            var status = ReadRegister(Register.STATUS_REG);
+            if (!((status & 0x04) == 1) && !((status & 0x08) == 1))
+            {
+                illuminance = new((ReadRegister(Register.LIGHT_REG_H) << 8 | ReadRegister(Register.LIGHT_REG_L)), UnitsNet.Units.IlluminanceUnit.Lux);
```

Comment:
You can use BinaryPrimitives for the conversion

Reply:
also in general if you decide to leave as is please at least add brackets - here it's not obvious whether `|` or `<<` has higher priority